scripts/commands/build.js

```diff
@@ -0,0 +1,86 @@
+import { exec } from 'child_process';
+import { promisify } from 'util';
+import { promises as fsPromises } from 'fs';
+import {
+	__dirname,
+	error,
+	success,
+	info,
+	getCurrentVersion,
+	getStatus,
+	isDevBuild,
+	pullLatestChanges,
+	verifyBuild,
+} from '../utils.js';
+
+const execAsync = promisify( exec );
+
+async function buildRelease() {
+	// Ensure we're always running in the project root.
+	process.chdir( `${ __dirname }/..` );
+
+	// Do a new release build
+
+	// Check to see if the working copy is clean
+	const status = await getStatus();
+	if ( ! status.isClean() ) {
+		error(
+			'Working copy is not clean. Please commit or stash your changes before releasing.'
+		);
+		return false;
+	}
+
+	// TODO - Verify (somehow) that we're not running with `start`.
+	/*
+	Some possible ideas for checking for `start`:
+
+	- Check for `build/*.map` files.
+	- The real build has a 948.js (not sure if this number changes) and start doesn't.
+	*/
+	if ( isDevBuild() ) {
+		error(
+			"You may have a dev build. Please make sure you aren't running 'npm start'"
+		);
+		return false;
+	}
+
+	await pullLatestChanges();
+	const version = getCurrentVersion();
+	info(
+		`Pulled latest changes from 'master'. Current version is ${ version }.`
+	);
+
+	await execAsync( 'npm i' );
```

Comment:
maybe it's useful to also check the node version based on [this file](https://github.com/Automattic/wc-calypso-bridge/blob/master/.nvmrc)?

Reply:
Excellent idea, thank you!